### PR TITLE
(GH-541) Check for latest PDK Version

### LINKED ISSUE #541: Check for latest PDK Version
**As a** user

**I would like to** have VSCode automatically check to see if my PDK install is up to date

**so that I** don't have to remember to do it myself.

### PR DESCRIPTION
**Pull request by jpogran**
Fixes #541

### CODE REVIEW

File: src/extension.ts [hunk 1]
```diff
@@ -231,3 +240,63 @@ async function notifyEditorServiceDisabled(context: vscode.ExtensionContext) {
     context.globalState.update(suppressEditorServicesDisabled, true);
   }
 }
+
+async function notifyIfNewPDKVersion(context: vscode.ExtensionContext, settings:IAggregateConfiguration) {
+  const suppressPDKUpdateCheck = 'suppressPDKUpdateCheck';
+  const dontCheckAgainNotice = "Don't check again";
+  const viewPDKDownloadPage = "More info";
+
+  // if (context.globalState.get(suppressPDKUpdateCheck, false)) {
```
Comment (glennsarti) [inline src/extension.ts]: nit: can probably delete this commented code

Comment (jpogran) [reply src/extension.ts]: Actually, that was supposed to be there. That honors the user telling us to stop checking for PDK versions. Although...there is no way to ask to be notified again...

Comment (glennsarti) [reply src/extension.ts]: >  Although...there is no way to ask to be notified again

That's not a usual workflow though. "Please pester me!"

File: src/extension.ts [hunk 2]
```diff
@@ -231,3 +240,63 @@ async function notifyEditorServiceDisabled(context: vscode.ExtensionContext) {
...
+async function notifyIfNewPDKVersion(context: vscode.ExtensionContext, settings:IAggregateConfiguration) {
+  const suppressPDKUpdateCheck = 'suppressPDKUpdateCheck';
+  const dontCheckAgainNotice = "Don't check again";
+  const viewPDKDownloadPage = "More info";
+
+  if (context.globalState.get(suppressPDKUpdateCheck, false)) {
+    return;
+  }
+
+  let version = '1.12.0';
```
Comment (glennsarti) [inline src/extension.ts]: I'm guessing this shouldn't be hardcoded?

**Comment by glennsarti** [comment]
out-of-scope: We should probably look at linting rules for spaces around statements to be consistent

`if(settings.ruby.pdkVersion){` vs `if (settings.ruby.pdkVersion) {`

**Comment by glennsarti** [comment]
Seems to work, however the message is a little odd. It asks to "click the more info button" but there is no more info button.

image

File: src/extension.ts (hunk 1 above)
Comment (jpogran) [reply src/extension.ts]: Uncommented and verified it works on my machine :)

File: src/extension.ts (hunk 2 above)
Comment (jpogran) [reply src/extension.ts]: Nope. Stupid mistake. fixed

**Comment by jpogran** [comment]
Message was odd because I left the 'lets add popups to both conditions so I can test either way' then 'i forgot to remove the extra condition'.

**Comment by glennsarti** [comment]
- [x] Confirm the more info button works as expected
- [x] Confirm message box is sane.
- [x] Confirm don't show this again works
- [x] Confirm enable/disable setting works

File: src/extension.ts (hunk 1 above)
Comment (glennsarti) [reply src/extension.ts]: This is still commented out.  Did you forget to push your changes?

**Pull request merged**